### PR TITLE
Add option to enable Java bindings in Open-MPI

### PR DESCRIPTION
**Pull request by galaxy001**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### CODE REVIEW

File: Library/Formula/open-mpi.rb [hunk 1]
```diff
@@ -29,6 +30,7 @@ class OpenMpi < Formula
 
   depends_on :fortran => :recommended
   depends_on "libevent"
+  depends_on :java if build.with? "mpi-java"
```
Comment (dunn) [inline Library/Formula/open-mpi.rb]: You can change this to `depends_on :java => :optional` and it'll create an option automatically (something like "Build with Java support"), then you don't need line 25 and line 49 can be `if build.with? "java"`.

**Comment by DomT4** [comment]
Was this the default prior to https://github.com/Homebrew/homebrew/pull/49704 or?

**Comment by galaxy001** [comment]
#49704 happens because the Java support here can only be enabled with `./configure --enable-mpi-java`.

So without `--enable-mpi-java`, Java is not required.

See also: <https://www.open-mpi.org/faq/?category=java#java_build>.

**Comment by galaxy001** [comment]
Hi, @DomT4

The Java binding is not enabled during default configuration. So I make it optional.

With `--enable-mpi-java`, it will generate `mpi.jar`, which is needed by other Java applets.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/afa67e702b1e214be7f0d8d0105ed6cacd34a44a. Thank you for another contribution to Homebrew @galaxy001! :smiley:

**Pull request closed without merging**